### PR TITLE
Implementing scoped model in your Flutter application

### LINKED ISSUE #648: Implementing scoped model in your Flutter application
### Brief Summary: 
The ScopedModel refers to the framework that allows data to flow from the parent widget down to its children or descendants. It allows users to build creative and highly interactive applications. Flutter is increasingly becoming popular today. It is, therefore, important for you as a developer to understand how the application receives and consumes data 

### Key Takeaways: 
- Understand the ScopedModel.
- Identify the required dependences to use the ScopedModel
- Know how to implement the ScopedModel in your Flutter application.

### References: 
https://pub.dev/packages/scoped_model#:~:text=A%20set%20of%20utilities%20that,when%20the%20model%20is%20updated.

### PR DESCRIPTION
**Pull request by WanjaMIKE**
This fixes #648 

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by hectorkambow** [comment]
👋  Good morning @WanjaMIKE  
Thank you for submitting your Flutter application article. We've started our preliminary revisions on this article. 
Upon running this article through our 3rd party plagiarism check - it raised a few red flags (mostly on the code snippets). 

 If you could please revisit the code sections to include all unique and different code snippets. We need to ensure we are bringing unique content to our community everytime (this includes both actual written content and code snippets). 👍  
Thank you v much. 
Looking forward to going through the rest of the article as it seems really helpful. 🚀

**Comment by WanjaMIKE** [comment]
Okay, Revising

**Comment by hectorkambow** [comment]
Thank you @WanjaMIKE  
I've also sent you another message via Slack FYI. 👍

File: articles/implementing-scoped-model-in-your-flutter-application/index.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+Open `Android Studio` click file, then create a `new Flutter project.` Ensure that you set the proper Flutter SDK path to avoid errors. The computer also needs to be online for Flutter dependencies and libraries to be installed. If this process is successful, your start page should look like the image below. 
+
+You have to be patient since this stage takes time.
+
+![Creating new project](/engineering-education/implementing-scoped-model-in-your-flutter-application/creating-project.png)
+
+### Installing the required library
+You must install the required library for you to access the scoped\_model's functionalities. Open the `pubspec.yaml` file, go to the dependencies section, and paste `scoped\_model : ^1.1.0`. Ensure that the statement aligns vertically with the term `flutter.` Ignoring this detail will result in errors during compilation. 
+
+Your `pubspec.yaml` file should look like the next image below.
```
Comment (hectorkambow) [inline articles/implementing-scoped-model-in-your-flutter-application/index.md:54]: @WanjaMIKE  
Is there an image missing here?

Comment (WanjaMIKE) [reply articles/implementing-scoped-model-in-your-flutter-application/index.md]: Yes. An image was missing. I have included the code snippet. Thanks

**Comment by hectorkambow** [comment]
@kayere Just finished up the prelim - if you could give the snippets a review for technical accuracy when you have a chance. 
Thank you Peter. 🚀

**Review by kayere — APPROVED** [review]
@hectorkambow - The article is good to go.

**Pull request merged**